### PR TITLE
Merge gilson-iot-rails4 into master

### PR DESCRIPTION
**Pull request by Ducz0r**
This should be thoroughly reviewed.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -26,7 +30,7 @@ gem 'uglifier', '>= 1.3.0'
 # jQuery & plugins
 gem 'jquery-turbolinks'
 gem 'jquery-rails'
-gem 'jquery-ui-rails'
+gem 'jquery-ui-rails', '~> 5.0'
```
Comment (okriuchykhin) [inline Gemfile]: I think it's not needed

File: Gemfile [hunk 2]
```diff
@@ -52,7 +56,7 @@ gem 'wicked_pdf'
 gem 'silencer' # Silence certain Rails logs
 gem 'wkhtmltopdf-heroku'
 gem 'remotipart', '~> 1.2' # Async file uploads
-gem 'faker' # Generate fake data
+gem 'faker', '~> 1.7.3' # Generate fake data
```
Comment (okriuchykhin) [inline Gemfile]: This one is also not needed

File: Gemfile [hunk 3]
```diff
@@ -92,9 +102,13 @@ group :production do
 end
 
 group :test do
-  gem 'minitest-reporters', '~> 1.1'
```
Comment (okriuchykhin) [inline Gemfile]: We should keep this

File: Gemfile [hunk 4]
```diff
@@ -92,9 +102,13 @@ group :production do
...
-  gem "shoulda-context"
```
Comment (okriuchykhin) [inline Gemfile]: This one as well

File: app/models/user.rb [hunk 5]
```diff
@@ -197,7 +199,18 @@ def name
   end
 
   def name=(name)
-    full_name = name
```
Comment (okriuchykhin) [inline app/models/user.rb]: I think we need keep this

File: app/views/user_notifications/_list.html.erb [hunk 6]
```diff
@@ -3,26 +3,9 @@
...
-        </div>
-        <% end %>
-        <% if notification.type_of == 'deliver' %>
-        <div class="text-center">
-          <span class="deliver"><%= fa_icon 'truck' %></span>
-        </div>
-        <% end %>
+        <%= render partial: "user_notifications/icons/#{notification.type_of}",
+                   locals: {notification: notification},
+                   formats: [:html] %>
```
Comment (okriuchykhin) [inline app/views/user_notifications/_list.html.erb]: Should we use this refactored version of notifications?

Comment (Ducz0r) [reply app/views/user_notifications/_list.html.erb]: What do you mean by refactored notifications? You're not referring to the Reactified, no?

Comment (okriuchykhin) [reply app/views/user_notifications/_list.html.erb]: No, just splitting notification types to different files. As far as I can remember it was done for easier inserts of new types, you just need to create partial with same name as notification type.

File: Gemfile (hunk 1 above)
Comment (Ducz0r) [reply Gemfile]: The version suffix, `'~ 5.0'`? Or the Gem itself?

File: Gemfile (hunk 2 above)
Comment (Ducz0r) [reply Gemfile]: The version suffix, `'~> 1.7.3'`? Or the Gem itself?

File: app/models/user.rb (hunk 5 above)
Comment (Ducz0r) [reply app/models/user.rb]: Instead of `self.full_name = name`?

Comment (okriuchykhin) [reply app/models/user.rb]: Yes, self is redundant here

File: Gemfile (hunk 2 above)
Comment (okriuchykhin) [reply Gemfile]: Version

File: Gemfile (hunk 1 above)
Comment (okriuchykhin) [reply Gemfile]: Version suffix

**Pull request merged**